x-pack/plugin/rank-rrf/src/main/java/org/elasticsearch/xpack/rank/rrf/RRFRankCoordinatorContext.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.rank.rrf;
+
+import org.apache.lucene.search.ScoreDoc;
+import org.apache.lucene.util.PriorityQueue;
+import org.elasticsearch.action.search.SearchPhaseController.SortedTopDocs;
+import org.elasticsearch.action.search.SearchPhaseController.TopDocsStats;
+import org.elasticsearch.common.util.Maps;
+import org.elasticsearch.search.SearchHit;
+import org.elasticsearch.search.query.QuerySearchResult;
+import org.elasticsearch.search.rank.RankCoordinatorContext;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import static org.elasticsearch.xpack.rank.rrf.RRFRankDoc.NO_RANK;
+
+/**
+ * Ranks and decorates search hits for RRF results on the coordinator.
+ */
+public class RRFRankCoordinatorContext extends RankCoordinatorContext {
+
+    private final int rankConstant;
+
+    public RRFRankCoordinatorContext(int size, int from, int windowSize, int rankConstant) {
+        super(size, from, windowSize);
+        this.rankConstant = rankConstant;
+    }
+
+    // used for faster hash lookup in a map of ranked documents
+    protected record RankKey(int doc, int shardIndex) {
+
+    }
+
+    @Override
+    public SortedTopDocs rank(List<QuerySearchResult> querySearchResults, TopDocsStats topDocsStats) {
```

Comment:
Could we have unit tests for this rank method?

Reply:
Added.